tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/EngSys/CodeownersTools/CodeownersTools.cs

```diff
@@ -251,10 +254,20 @@ private async Task<List<string>> CreateCodeownersPR(
             string description, // used for commit message, PR title, and PR description
             string branchPrefix,
             string identifier,
-            string workingBranch)
+            string workingBranch,
+            string repoOwner)
         {
             List<string> resultMessages = new();
             var branchName = "";
+            var hasPushPermissions = await githubService.HasPushPermission(repoOwner, repo);
+
+            if (!hasPushPermissions)
+            {
+                var originalRepoOwner = repoOwner; // Store for messaging
+                repoOwner = Constants.AZURE_OWNER_PATH;
+                workingBranch = "";
+                resultMessages.Add($"GitHub token does not have permission to push to {repoOwner} repository, opening a new PR on a branch in the main repo");
```

Comment:
If you move `resultMessages.Add` up to before you set `repoOwner = Constants.AZURE_OWNER_PATH` then you don't have to allocate a new variable.